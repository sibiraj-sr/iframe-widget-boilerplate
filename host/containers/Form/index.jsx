import React from 'react';

import Label from '../../components/Label';

function Form () {
  return (
    <div className='w-full pt-12 px-6 pb-8 bg-gray-300'>
      <div className='form-header mb-4'>
        <p className='text-4xl font-bold'>Write to us</p>
        <p className='font-bold leading-relaxed'>
          Tell us what you are experiencing. The more details you give us, the faster we can assist you.
        </p>
      </div>
      <div className='form-content'>
        <form className='w-full max-w-lg'>
          <div className='flex flex-wrap mb-4'>
            <div className='w-1/2 pr-3'>
              <Label required>First Name</Label>
              <input type='text' placeholder='John' className='w-full rounded pl-2 text-sm h-8' maxLength='25' />
            </div>
            <div className='w-1/2 pl-3'>
              <Label required>Last Name</Label>
              <input type='text' placeholder='Doe' className='w-full rounded pl-2 text-sm h-8' maxLength='25' />
            </div>
          </div>
          <div className='mb-4'>
            <Label required>I would like help with</Label>
            <textarea className='w-full resize-none' rows='8' maxLength='400' />
          </div>
        </form>
      </div>
    </div>
  );
}

export default Form;
