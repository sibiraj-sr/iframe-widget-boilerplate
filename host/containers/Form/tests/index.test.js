import React from 'react';
import { render } from '@testing-library/react';

import Form from '../index';

describe('<Form />', () => {
  it('should render and match the snapshot', () => {
    const { asFragment } = render(<Form />);
    const renderedOutput = asFragment();
    expect(renderedOutput).toMatchSnapshot();
  });
});
