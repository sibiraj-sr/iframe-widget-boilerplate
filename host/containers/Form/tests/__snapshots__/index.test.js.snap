// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> should render and match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="w-full pt-12 px-6 pb-8 bg-gray-300"
  >
    <div
      class="form-header mb-4"
    >
      <p
        class="text-4xl font-bold"
      >
        Write to us
      </p>
      <p
        class="font-bold leading-relaxed"
      >
        Tell us what you are experiencing. The more details you give us, the faster we can assist you.
      </p>
    </div>
    <div
      class="form-content"
    >
      <form
        class="w-full max-w-lg"
      >
        <div
          class="flex flex-wrap mb-4"
        >
          <div
            class="w-1/2 pr-3"
          >
            <label
              class="block mb-1 text-sm"
            >
              <span
                class="inline-block text-red-500 mr-1"
              >
                *
              </span>
              First Name
            </label>
            <input
              class="w-full rounded pl-2 text-sm h-8"
              maxlength="25"
              placeholder="John"
              type="text"
            />
          </div>
          <div
            class="w-1/2 pl-3"
          >
            <label
              class="block mb-1 text-sm"
            >
              <span
                class="inline-block text-red-500 mr-1"
              >
                *
              </span>
              Last Name
            </label>
            <input
              class="w-full rounded pl-2 text-sm h-8"
              maxlength="25"
              placeholder="Doe"
              type="text"
            />
          </div>
        </div>
        <div
          class="mb-4"
        >
          <label
            class="block mb-1 text-sm"
          >
            <span
              class="inline-block text-red-500 mr-1"
            >
              *
            </span>
            I would like help with
          </label>
          <textarea
            class="w-full resize-none"
            maxlength="400"
            rows="8"
          />
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
